app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/CachedFirestoreDatabaseManagement.kt

```diff
@@ -0,0 +1,69 @@
+package com.github.HumanLearning2021.HumanLearningApp.firestore
+
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.Converters
+import com.github.HumanLearning2021.HumanLearningApp.offline.CachePictureRepository
+import com.github.HumanLearning2021.HumanLearningApp.offline.OfflineCategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.offline.PictureRepository
+
+class CachedFirestoreDatabaseManagement internal constructor(
+    val dbName: String
+): FirestoreDatabaseManagement(FirestoreDatabaseService(dbName)) {
+
+    private lateinit var cache: PictureRepository
+    private val cachedPictures: MutableMap<String, FirestoreCategorizedPicture> = mutableMapOf()
+
+    /**
+     * Should be called before the cached database management is first used (in onCreate for activities)
+     * @param context: the context of the calling application
+     */
+    fun initialize(context: Context): CachedFirestoreDatabaseManagement {
```

Comment:
`@dagger.hilt.android.qualifiers.ApplicationContext` can be use to inject and avoid this dance

Reply:
Done

Reply:
That method is still there it should be gone